src/Microsoft.OData.Edm/Schema/EdmModelBase.cs

```diff
@@ -173,25 +198,21 @@ public virtual IEnumerable<IEdmOperation> FindDeclaredBoundOperations(IEdmType b
         /// </returns>
         public virtual IEnumerable<IEdmOperation> FindDeclaredBoundOperations(string qualifiedName, IEdmType bindingType)
         {
-            IEnumerable<IEdmOperation> enumerable = this.FindDeclaredOperations(qualifiedName);
-            IList<IEdmOperation> operations = enumerable as IList<IEdmOperation>;
+            IList<IEdmOperation> operations = this.FindDeclaredBoundOperations(bindingType) as IList<IEdmOperation>;

```

Comment:
> as IList<IEdmOperation> [](start = 91, length = 24)

since FindDeclaredBoundOperations is virtual, probably can't assume it is an IList -- instead, try casting and, if that fails, fall back to enumerable logic.